core/templates/pages/learner-dashboard-page/new-learner-dashboard-page/home-tab.component.ts

```diff
@@ -0,0 +1,79 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Component for home tab in the Learner Dashboard page.
+ */
+
+import { Component, EventEmitter, Input, Output } from '@angular/core';
+import { LearnerTopicSummary } from 'domain/topic/learner-topic-summary.model';
+import { LearnerDashboardPageConstants } from 'pages/learner-dashboard-page/learner-dashboard-page.constants';
+import { DeviceInfoService } from 'services/contextual/device-info.service';
+
+ @Component({
+   selector: 'oppia-home-tab',
+   templateUrl: './home-tab.component.html'
+ })
+export class HomeTabComponent {
+  @Output() setActiveSection: EventEmitter<string> = new EventEmitter();
+  @Input() currentGoals: LearnerTopicSummary[];
+  @Input() untrackedTopics: Record<string, LearnerTopicSummary[]>;
+  @Input() username: string;
+  nextIncompleteNodeTitles: string[] = [];
+  widthConst: number = 233;
+  width: number;
+
+  constructor(
+    private deviceInfoService: DeviceInfoService
+  ) {}
+
+  ngOnInit(): void {
+    this.width = this.widthConst * (this.currentGoals.length);
+  }
+
+  getTimeOfDay(): string {
+    let time = new Date().getHours();
+
+    if (time <= 12) {
+      return 'morning';
+    } else if (time <= 18) {
+      return 'afternoon';
+    }
+    return 'evening';
+  }
+
+  isNonemptyObject(object: Object): boolean {
+    return Object.keys(object).length !== 0;
+  }
+
+  getClassroomLink(classroomUrlFragment: string): string {
+    return `/learn/${classroomUrlFragment}`;
+  }
+
+  checkMobileView(): boolean {
+    return this.deviceInfoService.isMobileDevice();
+  }
+
+  getWidth(length: number): number {
+    if (length >= 3) {
+      return 662;
+    }
+    return (length + 1) * 164;
```

Comment:
Can you leave a comment on how or why these numbers were taken?

Reply:
Done.